app/src/views/OperationalLearning/Stats/index.tsx

```diff
@@ -0,0 +1,352 @@
+const regionLabelSelector = (datum: RegionStatItem) => datum.region_name;
+
+type SourceType = 'dref' | 'emergencyAppeal' | 'others';
+interface SourceTypeOption {
+    key: SourceType;
+    label: string;
+    color: string;
+}
+
+const dataKeys: SourceType[] = [
+    'dref',
+    'emergencyAppeal',
+    'others',
+];
+const dataKeyToClassNameMap: Record<SourceType, string> = {
+    dref: styles.dref,
+    emergencyAppeal: styles.emergencyAppeal,
+    others: styles.others,
+};
+const sourceClassNameSelector = (dataKey: SourceType) => dataKeyToClassNameMap[dataKey];
+const xAxisFormatter = (date: Date) => date.toLocaleString(
+    navigator.language,
+    { year: 'numeric' },
+);
+
+type SourceTypeEnum = components<'read'>['schemas']['ApiAppealTypeEnumKey'];
+const SOURCE_TYPE_EMERGENCY = 1 satisfies SourceTypeEnum;
+const SOURCE_TYPE_DREF = 0 satisfies SourceTypeEnum;
+
+const transformSourcesOverTimeData = (data: SourcesOverTimeItem[]) => {
+    const groupedData: Record<string, Record<SourceType, number>> = {};
+
+    data.forEach((entry) => {
+        const year = getFormattedDateKey(entry.date);
+        if (!groupedData[year]) {
+            groupedData[year] = { dref: 0, emergencyAppeal: 0, others: 0 };
+        }
+        if (entry.atype === SOURCE_TYPE_DREF) {
+            groupedData[year].dref += entry.count;
+        } else if (entry.atype === SOURCE_TYPE_EMERGENCY) {
+            groupedData[year].emergencyAppeal += entry.count;
+        } else {
+            groupedData[year].others += entry.count;
+        }
+    });
+
+    return groupedData;
+};
+
+interface Props {
+    query: OpsLearningQuery | undefined
+}
+
+function Stats(props: Props) {
+    const {
+        query,
+    } = props;
+
+    const strings = useTranslation(i18n);
+    const alert = useAlert();
+    const [activePointKey, setActivePointKey] = useState<string>();
+
+    const {
+        response: learningStatsResponse,
+        pending: learningStatsPending,
+    } = useRequest({
+        url: '/api/v2/ops-learning/stats/',
+        query,
+        onFailure: () => {
+            alert.show(
+                strings.failedToFetchStats,
+                { variant: 'danger' },
+            );
+        },
+    });
+
+    const sourcesOverTimeData = useMemo(
+        () => {
+            if (isNotDefined(learningStatsResponse)) {
+                return undefined;
+            }
+            return transformSourcesOverTimeData(learningStatsResponse.sources_overtime);
+        },
+        [learningStatsResponse],
+    );
+
+    const dateList = useMemo(() => {
+        if (isNotDefined(sourcesOverTimeData)) {
+            return undefined;
+        }
+        const dates = Object.keys(sourcesOverTimeData).map((year) => new Date(year));
+        const oldestDate = new Date(Math.min(...dates.map((date) => date.getTime())));
+        const latestDate = new Date(Math.max(...dates.map((date) => date.getTime())));
```

Comment:
We should also check for length of dates

Reply:
Done.